### PR TITLE
Add Packit support with smoke test

### PR DESCRIPTION
**Pull request by jkonecny12**
This will just make a simple smoke test for initial-setup with implementation for local execution. Latest Anaconda will always be installed before because initial-setup heavily depends on Anaconda.

It should be pretty easy to add pylint when the code is ready.

**IDEA**
It could be interesting to avoid using mock and use `packit copr-build` instead. However, I will take longer (because of COPR queue) and also it does not work for me for some reason. I'll try to investigate why.

### CODE REVIEW

**Review by poncovka — APPROVED** [review]
Looks good to me. Thanks!

**Comment by poncovka** [comment]
> It should be pretty easy to add pylint when the code is ready.

Please, open a separate pull request for pylint once this one is merged. Otherwise, the code will never be ready.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Because of `tar` fail on Packit service I've took this as opportunity and updated the archive creation to `setup.py` solution. We are using this on `simpleline` and also on `dasbus` so it's more consistency.
* When I was doing that I've tidy the setup.py
** Martin Kolman is now maintainer (it's about time) :)
** Fix URL of the project
** Use python3 instead of python2
* Remove `make test` from the spec file `%check` call

**Comment by jkonecny12** [comment]
@M4rtinK a few questions on you.

Could you please test if this correctly works? My testing went fine but still.
Do you want to keep `Makefile` `local` command? I think it's not really useful anymore but I want your acknowledge first.

Also the local test run is not really nice but it works correctly and it's fortunately not that long to run but it would be nice to have also not TMT solution for the quick tests run.

**Comment by jkonecny12** [comment]
Hello @thrix, I wonder, is the above test failure on `rawhide` because of bug in ``Cruncher`` or is the problem on my side? I'm able to run the test correctly locally.

Error message:
```
 No tests were defined in FMF metadata. 
```

However, in the log output:
```
[15:37:12] [+] [cruncher] Discovered testsets:
[ "/plans/smoke"]
[15:37:12] [+] [cruncher] [/plans/smoke] Summary 'Basic smoke test'
[15:37:12] [+] [cruncher] [/plans/smoke] Testset working directory '/opt/cruncher/artifacts/32debe17-d0b4-43fa-9ec8-4308311f7c96/plans-smoke'
[15:37:12] [+] [cruncher] [/plans/smoke] [discover] Discovered tests:
[ "/tests/smoke"]
```

**Comment by jkonecny12** [comment]
**UPDATED:**
* Switch ELN for just a build instead of test run.

File: Makefile
Comment (martinpitt) [inline Makefile:69]: This is unnecessary.

File: initial-setup.spec
Comment (martinpitt) [inline initial-setup.spec:138]: I wasn't aware that this is still some Fedora best practice, in times of web accessible git repos.. But sure! Indeed on my system there are 108 /usr/share/doc/*/ChangeLog files.

**Review by martinpitt — APPROVED** [review]
Looks great, thank you!

File: Makefile
Comment (jkonecny12) [reply Makefile]: Fixed.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fixed improvement pointed by @martinpitt

**Review by poncovka — APPROVED** [review]
Looks good to me.

**Comment by thrix** [comment]
> Hello @thrix, I wonder, is the above test failure on `rawhide` because of bug in `Cruncher` or is the problem on my side? I'm able to run the test correctly locally.
> 
> Error message:
> 
> ```
>  No tests were defined in FMF metadata. 
> ```
> 
> However, in the log output:
> 
> ```
> [15:37:12] [+] [cruncher] Discovered testsets:
> [ "/plans/smoke"]
> [15:37:12] [+] [cruncher] [/plans/smoke] Summary 'Basic smoke test'
> [15:37:12] [+] [cruncher] [/plans/smoke] Testset working directory '/opt/cruncher/artifacts/32debe17-d0b4-43fa-9ec8-4308311f7c96/plans-smoke'
> [15:37:12] [+] [cruncher] [/plans/smoke] [discover] Discovered tests:
> [ "/tests/smoke"]
> ```

it is in cruncher, it will get resolved once we migrate to the new backend finally

**Comment by jkonecny12** [comment]
**UDPATED:**
* Switch to test from source not RPM (much faster)
* Re-execution of test is much much more quicker now because container is provisioned just once and prepared just once
* Use not-yet-released adjust solution of TMT to make configuration of source/rpm test much easier and cleaner.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Remove old debugging leftovers
* Add custom Anaconda repo implementation -- great way to test your changes there
* Add README note

**Review by poncovka — APPROVED** [review]
It looks great!

File: tests/smoke.fmf
Comment (jkonecny12) [reply tests/smoke.fmf:11]: Not sure what do you mean by ``harmless subdir``. The adjust condition will take a place just when we have RPM file pre-installed which is happening on Packit. In that case we are getting to other place than project root to avoid import of the local ``initial_setup`` folder but instead use the system one.

**Review by M4rtinK — APPROVED** [review]
Looks good to me & seems to work fine locally as well & nice cleanup. Thanks! :)

File: tests/smoke.fmf
Comment (martinpitt) [reply tests/smoke.fmf:11]: I suppose I misunderstood. I thought you want to *always* run the test in a directory which does *not* have the local source package with its python module directory, so that the python import will happen from the system-installed RPM path. But apparently you actually do want to test the git checkout in the "other" non-rpm case (whatever that is)?

Comment (jkonecny12) [reply tests/smoke.fmf:11]: It's local. I'll try to improve the comment there. Seems it's not much helpful.

In general, when TMT is started by Packit/Gating it will also provide built package to TMT and this is installed to the system. You should always test these instead of source. However, when you run it locally (``make test``) then there is no automation to provide you this package, also it would take too much time when you need to build RPM for each test run. Because of that I'm adding the ``adjust``.

``adjust`` implementation in this case will change the path to not test source but RPM in case when it's started from Packit. That happenend thanks to the [trigger code](https://tmt.readthedocs.io/en/latest/spec/context.html#trigger).

**Comment by jkonecny12** [comment]
**UPDATED:**
* Improve comment about adjust in the smoke test FMF file.

**Comment by martinpitt** [comment]
Thanks!

**Pull request merged**
